InternalDocs/garbage_collector.md

```diff
@@ -351,38 +351,77 @@ follows these steps in order:
-survives a collection of its generation it will be moved to the next one
-(generation 1), where it will be surveyed for collection less often. If
-the same object survives another GC round in this new generation (generation 1)
-it will be moved to the last generation (generation 2) where it will be
-surveyed the least often.
-
-The GC implementation for the free-threaded build does not use multiple
-generations.  Every collection operates on the entire heap.
+two generations: young and old. Every new object starts in the young generation.
+In order to keep pause times down, scanning of the old generation of the heap
+occurs in increments. To keep track of what has been scanned,
+the old generation contains two lists: scanned and unscanned heap.
+
+To detect and collect all unreachable objects in the heap, the garbage collector
+must scan the whole heap. This whole heap scan is called a full scavenge.
+
+To limit the time each garbage collection takes, the detection and collection
+algorithm is executed only on a portion of the heap called an increment.
+For each full scavenge, the increments will cover the whole heap.
+
+Each increment, the portion of the heap scanned by a single collection is made up
+of three parts:
+
+* The young generation
+* The old generation's least recently objects
+* All objects reachable from those objects that have not yet been scanned this full scavenge
+
+Any young generation objects surviving this collection are moved to the old generation,
+and reachable objects in the old generation remain in the old generation.
+The old generation is composed of two lists: scanned and unscanned.
+(The implementation refers to the unscanned part as `pending` and the scanned part
+as `visited`).
+Survivors are moved to the back of the scanned list. The old part of increment is taken
+from the front of the unscanned list.
+
+When a full scavenge starts, no objects in the heap are considered to have been scanned.
+When all objects in the heap have been scanned a cycle ends, and all objects are
+considered unscanned again.
+
+In order to collect all unreachable cycles, each increment must contain all of
+an unreachable cycle, or none of it.
+In order to make sure that the whole of any unreachable cycle is contained in an
+increment,  all unscanned objects reachable from any object in the increment must
+be included in the increment.
+Thus, to form a complete increment we perform a
+[transitive closure](https://en.wikipedia.org/wiki/Transitive_closure)
+over reachable, unscanned objects from the initial increment.
+We can exclude scanned objects, as they must have been reachable when scanned.
+If a scanned object becomes part of an unreachable cycle after being scanned, it
+will not be collected this cycle, but it will be collected next full scavenge.
+
+.. note::
+
+   The GC implementation for the free-threaded build does not use incremental collection.
+   Every collection operates on the entire heap.
```

Comment:
Per GitHub Markdown syntax https://docs.github.com/en/get-started/writing-on-github/getting-started-with-writing-and-formatting-on-github/basic-writing-and-formatting-syntax#alerts
```suggestion
> [!NOTE]
> The GC implementation for the free-threaded build does not use incremental collection.
> Every collection operates on the entire heap.
```